torch/ao/quantization/_pt2e/quantizer/qnnpack_quantizer.py

```diff
@@ -458,74 +460,54 @@ def _annotate_conv2d(
-                )
-            if _is_annotated([act_use_node]) is False:  # type: ignore[list-item]
-                _annotate_input_qspec_map(
-                    act_use_node, act_node, get_act_obs_or_fq_ctr(quantization_config)
-                )
-            if bias_node and _is_annotated([bias_node]) is False:
-                _annotate_output_qspec(
-                    bias_node, get_bias_obs_or_fq_ctr(quantization_config)
-                )
-            if _is_annotated([weight_node]) is False:  # type: ignore[list-item]
-                _annotate_output_qspec(
-                    weight_node, get_weight_obs_or_fq_ctr(quantization_config)
-                )
-            if _is_annotated([output_node]) is False:
-                _annotate_output_qspec(
-                    output_node, get_act_obs_or_fq_ctr(quantization_config)
-                )
-            nodes_to_mark_annotated = list(match.nodes_map.values())
-            nodes_to_mark_annotated.remove(act_node)
-            _mark_nodes_as_annotated(nodes_to_mark_annotated)
+        module_partitions = get_source_partitions(
+            gm.graph, [torch.nn.Linear, torch.nn.functional.linear]
+        )
+        for module_or_fn_type, partitions in module_partitions.items():
+            if module_or_fn_type == torch.nn.Linear:
+                for p in partitions:
+                    act_node = p.input_nodes[0]
+                    output_node = p.output_nodes[0]
+                    weight_node = None
+                    bias_node = None
+                    for node in p.params:
+                        weight_or_bias = getattr(gm, node.target)  # type: ignore[arg-type]
+                        if weight_or_bias.ndim == 2:  # type: ignore[attr-defined]
+                            weight_node = node
+                        if weight_or_bias.ndim == 1:  # type: ignore[attr-defined]
+                            bias_node = node
+                    if weight_node is None:
+                        raise ValueError("No weight found in Linear pattern")
+                    # find use of act node within the matched pattern
+                    act_use_node = None
+                    for node in p.nodes:
+                        if node in act_node.users:
+                            act_use_node = node
+                            break
+                    if act_use_node is None:
+                        raise ValueError(
+                            "Could not find an user of act node within matched pattern."
+                        )
+                    if _is_annotated([act_use_node]) is False:  # type: ignore[list-item]
+                        _annotate_input_qspec_map(
+                            act_use_node,
+                            act_node,
+                            get_act_obs_or_fq_ctr(quantization_config),
+                        )
+                    if bias_node and _is_annotated([bias_node]) is False:
+                        _annotate_output_qspec(
```

Comment:
are we still annotating output? or do you plan to change it to input_qspec_map?

Reply:
are you talking about bias quantization using derived quantization? Otherwise it doesnt really matter? For weight quantization this PR still uses output annotation